eZ/Publish/Core/FieldType/RelationList/Type.php

```diff
@@ -257,6 +261,21 @@ public function validate(FieldDefinition $fieldDefinition, SPIValue $fieldValue)
             );
         }
 
+        foreach ($fieldValue->destinationContentIds as $destinationContentId) {
+            try {
+                $this->contentService->loadContent($destinationContentId);
+            } catch (NotFoundException | UnauthorizedException $e) {
+                $validationErrors[] = new ValidationError(
+                    'Content with identifier %contentId% is not a valid relation target',
+                    null,
+                    [
+                        '%contentId%' => $destinationContentId,
+                    ],
+                    'destinationContentId'
+                );
+            }
```

Comment:
This part is repeatable, so should be extracted to a separate Validator

Reply:
Resolved in https://github.com/ezsystems/ezpublish-kernel/pull/3131/commits/53c034f38a8932ad8c3ced9f8cb0058e5b6add9b.

Reply:
Please note that we cannot use the 'standard' `Validator` class here as `Value` differs in each fieldtype, so looks a bit like overkill.